mfkdf2/src/definitions/mfkdf_derived_key/mod.rs

```diff
@@ -64,3 +67,28 @@ impl std::fmt::Display for MFKDF2DerivedKey {
     )
   }
 }
+
+impl MFKDF2DerivedKey {
+  /// Derive an internal key for deriving separate keys for parameters, secret, and integrity
+  /// using the policy salt and secret.
+  fn derive_internal_key(&self) -> MFKDF2Result<Key> {
```

Comment:
this should be infallible, no?

Not a big deal but it would be good to have functions that can't panic and don't return errors!

Perhaps we should just make this a general issue to tackle.